client/extensions/woocommerce/state/wc-api/products/actions.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Internal dependencies
+ */
+import wp from 'lib/wp';
+import { error } from '../actions';
+import {
+	WOOCOMMERCE_API_CREATE_PRODUCT,
+	WOOCOMMERCE_API_CREATE_PRODUCT_SUCCESS,
+} from '../../action-types';
+
+export function createProduct( siteId, product ) {
+	return ( dispatch ) => {
+		const createAction = {
+			type: WOOCOMMERCE_API_CREATE_PRODUCT,
+			payload: { siteId, product },
+		};
+
+		dispatch( createAction );
+
+		// Filter out any id we might have.
+		const { id, ...productData } = product;
+
+		if ( typeof id === 'number' ) {
+			dispatch( error( siteId, createAction, {
+				message: 'Attempting to create a product which already has a valid id.',
+				product,
+			} ) );
+			return;
+		}
+
+		const jetpackProps = { path: `/jetpack-blogs/${ siteId }/rest-api/` };
+		const httpProps = {
+			path: '/wc/v2/products',
+			body: JSON.stringify( productData ),
+			json: true,
+		};
+
+		return wp.req.post( jetpackProps, httpProps )
```

Comment:
From what I know, Calypso is moving to making network requests using the [data layer](https://github.com/Automattic/wp-calypso/tree/master/client/state/data-layer). This is so that we can offer offline support & better managing multiple requests.

I just did a migration to the data layer for another component, so would be happy to help.

cc @dmsnell for a more nuanced opinion.